packages/@aws-cdk/aws-msk-alpha/test/cluster.test.ts

```diff
@@ -785,5 +787,65 @@ describe('MSK Cluster', () => {
       });
     });
   });
+
+  describe('created with storage mode', () => {
+    describe('with tiered storage mode', () => {
+      test('fails if incompatible Kafka version', () => {
+        expect(
+          () =>
+            new msk.Cluster(stack, 'Cluster', {
+              clusterName: 'cluster',
+              kafkaVersion: msk.KafkaVersion.V2_6_1,
+              vpc,
+              storageMode: msk.StorageMode.TIERED,
+            }),
+        ).toThrow(
+          'To utilize Tiered storage mode, the MSK cluster Kafka version must be compatiable.',
+        );
+      });
+
+      test('fails if instance type is t3.small', () => {
+        expect(
+          () =>
+            new msk.Cluster(stack, 'Cluster', {
+              clusterName: 'cluster',
+              instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.SMALL),
+              kafkaVersion: msk.KafkaVersion.V2_8_2_TIERED,
+              vpc,
+              storageMode: msk.StorageMode.TIERED,
+            }),
+        ).toThrow(
+          'The t3.small instance type does not support Tiered storage mode.',
+        );
+      });
+
+      test('create a cluster with tiered storage mode', () => {
```

Comment:
Can you please [move this test above](https://github.com/aws/aws-cdk/pull/27560/files#r1367765991) the `fails if incompatible Kafka version` one?